src/core/common/message.cpp

```diff
@@ -80,6 +80,8 @@ Message *MessagePool::Allocate(Message::Type aType, uint16_t aReserveHeader, con
     message->SetType(aType);
     message->SetReserved(aReserveHeader);
     message->SetLinkSecurityEnabled(aSettings.IsLinkSecurityEnabled());
+    message->SetAllowLoopbackToHost(true);
```

Comment:
Should we set the default value for this based on `OPENTHREAD_CONFIG_IP6_ALLOW_LOOP_BACK_HOST_DATAGRAMS` config?

Reply:
I don't have much knowledge about the use case of `OPENTHREAD_CONFIG_IP6_ALLOW_LOOP_BACK_HOST_DATAGRAMS`. Previously it was only used in `otIp6Send()` and `Nat64::Translator::SendMessage()` (I previously checked with Song and he said it may not be necessary.)

So I wonder if we still need this config? In the previous code, most of the cases we're explicitly setting it to true or false (when calling `HandleDatagram()`) regardless of this config's value.

Reply:
We should definitely keep this configuration. It addresses a possible "ping-pong" issue that can occur on devices (that may not be BRs). I know we faced this issue in some of our internal products, and I think other teams and products may be using this configuration as well (to address same situation).

The issue occurs when the host network IP stack decides to send an IP message to the OT stack to be sent over Thread, but the OT stack detects that the destination is not reachable and passes the message back to the host. This 
 process then repeats. In our products, this situation happened on on some devices (not acting as BR) running an RTOS with "lwip" as the host network stack.

Setting `OPENTHREAD_CONFIG_IP6_ALLOW_LOOP_BACK_HOST_DATAGRAMS` to false ensures that any message given to the OT stack using `otIp6Send()` is never passed back to the host. This is why it was used in this API itself

Reply:
Thanks for sharing the context! I've set `OPENTHREAD_CONFIG_IP6_ALLOW_LOOP_BACK_HOST_DATAGRAMS` as the default value when creating a message.